### DISCUSSION TITLE
Example JSON to inject Please

### DISCUSSION THREAD

**Discussion by MasterJames** [quietvoid/hdr10plus_tool#74] [Q&A]
Hi I am having trouble figuring out what the JSON formatting is like to inject the "Dynamic Metadata" (with scene or frame range MaxCLL MaxFALL or whatever it might be maybe "MasteringDisplay_Luminance":"min: 0.0001 cd/m2, max: 1000 cd/m2", but what the range vales are if it goes in a properties folder or what? Or where would I get that from. I tried to extract some from a YouTube video that claims to have it but I guess what I downloaded via yt-dlp does get anything because it says "File doesn't contain dynamic metadata". The process was that I used ffmpeg to convert a webm to mkv but still nothing.

Also tried exporting a timeline's JSON from Davinci Resolve but it fails to have anything meaningful in the file exported as well.

**Comment by quietvoid** [comment]
There are examples in the repo: https://github.com/quietvoid/hdr10plus_tool/blob/main/assets/hevc_tests/regular_metadata.json

Currently `hdr10plus_tool` only supports HEVC files, so it won't work for VP9/webm.

**Reply by MasterJames** [reply]
Okay thanks I had tried to convert from it to MKV first. Alas it did not transfer that way. Eventually I figured how to get Resolve to Analyze and then a timeline will export only per frame (?! seemed ridiculous the spec didn't allow for frame ranges.) SD down conversion guides. Currently Resolve version has been downgraded to only allow an un-editable visualization and looks like an annoying bug that it can't be removed after the fact [so a separate backup file is required then]) and just seems like default linear values. Anyway it looks like this in JSON 
```

{
    "JSONInfo": {
        "DateTimeJSONLastModified": "2024-02-29T07:16:53Z",
        "HDR10plusProfile": "A",
        "Version": "1.1"
    },
    "SceneInfo": [
        {
            "DateTimeMetadataGenerated": "2024-02-29T07:16:53Z",
            "LuminanceParameters": {
                "AverageRGB": 415,
                "LuminanceDistributions": {
                    "DistributionIndex": [
                        1,
                        5,
                        10,
                        25,
                        50,
                        75,
                        90,
                        95,
                        99
                    ],
                    "DistributionValues": [
                        1,
                        3574,
                        93,
                        103,
                        221,
                        495,
                        708,
                        1326,
                        3574
                    ]
                },
                "MaxScl": [
                    0,
                    0,
                    0
                ]
            },
            "NumberOfWindows": 1,
            "SceneFrameIndex": 0,
            "SceneId": 0,
            "SequenceFrameIndex": 108000,
            "TargetedSystemDisplayMaximumLuminance": 0
        },
repeat for all frames, yuck.
,
        {
            "DateTimeMetadataGenerated": "2024-02-29T07:16:53Z",
            "LuminanceParameters": {
                "AverageRGB": 442,
                "LuminanceDistributions": {
                    "DistributionIndex": [
                        1,
                        5,
                        10,
                        25,
                        50,
                        75,
                        90,
                        95,
                        99
                    ],
                    "DistributionValues": [
                        1,
                        3574,
                        93,
                        103,
                        221,
                        495,
                        708,
                        1327,
                        3574
                    ]
                },
                "MaxScl": [
                    0,
                    0,
                    0
                ]
            },
            "NumberOfWindows": 1,
            "SceneFrameIndex": 268,
            "SceneId": 1,
            "SequenceFrameIndex": 108538,
            "TargetedSystemDisplayMaximumLuminance": 0
        }
    ],
    "SceneInfoSummary": {
        "SceneFirstFrameIndex": [
            108000,
            108270
        ],
        "SceneFrameNumbers": [
            270,
            269
        ]
    },
    "TitleInfo": {
        "AlternateVersion": "Theatrical",
        "Title": "Timeline 1"
    }
}

```
Which I see is what your link has so thanks for that, so just a few questions for clarity if you or someone in discussion could explain.
I see the SceneFrameNumbers is a total number of Frames per scene, so I figured that out.
I noticed MaxScl and TargetedSystemDisplayMaximumLuminance sometimes have a value of zero "0" (three in the case of MaxScl) is that a meaning like default or unlimited aka infinity? What does the Key MaxScl mean or is for?
Oh I did find something about that here then https://ffmpeg.org/doxygen/trunk/structAVHDRPlusColorTransformParams.html#a291012e029b09282caad1c1399e0b1b8

```
The maximum of the color components of linearized RGB values in the processing window in the scene.

The values should be in the range of 0 to 1, inclusive and in multiples of 0.00001. maxscl[ 0 ], maxscl[ 1 ], and maxscl[ 2 ] are corresponding to R, G, B color components respectively.

```
Your sample file has values like this "MaxScl": [ 17513, 16895, 14316 ] but the spec in code I found says it's suppose to be between zero and one.
Which is right or wrong and so any information here would be clarifying and appreciated. Despite the definition I still can't say clearly what it is used for or represents? or what happens if it's zero (I'm hoping it's undefined and recalculated by interpreter because Resolve did that in my sample as you can see).

Further researching and looking at the definition I went and found a definition for 'inclusive' ...

`Inclusive in math means that the starting and ending numbers are included in the set or range. For example, if we say "count from 1 to 10 inclusively," we would include the numbers 1 and 10 in the set. On the other hand, exclusive in math means that the starting and ending numbers are not included in the set or range.
`
Not sure if it's a running total and so over 1 makes sense now either.

Thanks for your good work and help with understanding.

**Reply by quietvoid** [reply]
The values in the JSON are linear nits (0 to 10 000 nits for HDR) multiplied by 10.
The range is 0 to 100 000.

You can look up the SMPTE ST 2094-40 specification.
